jpa/hibernate5/src/main/java/org/jboss/as/jpa/hibernate5/HibernateSecondLevelCache.java

```diff
@@ -88,41 +79,39 @@ public static void addSecondLevelCacheDependencies(Properties mutableProperties,
-            cacheSettings.put(NATURAL_ID, mutableProperties.getProperty(NATURAL_ID_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
-            if (mutableProperties.getProperty(PENDING_PUTS_CACHE_RESOURCE_PROP) != null) {
-                cacheSettings.put(PENDING_PUTS, mutableProperties.getProperty(PENDING_PUTS_CACHE_RESOURCE_PROP));
-            }
-            if (Boolean.parseBoolean(mutableProperties.getProperty(AvailableSettings.USE_QUERY_CACHE))) {
-                cacheSettings.put(QUERY, mutableProperties.getProperty(QUERY_CACHE_RESOURCE_PROP, DEF_QUERY_RESOURCE));
-                cacheSettings.put(TIMESTAMPS, mutableProperties.getProperty(TIMESTAMPS_CACHE_RESOURCE_PROP, DEF_QUERY_RESOURCE));
-            }
-
-            // Collect distinct cache configurations for standard regions
-            Set<String> standardRegionConfigs = Stream.of(ENTITY, IMMUTABLE_ENTITY, COLLECTION, NATURAL_ID, PENDING_PUTS, QUERY, TIMESTAMPS)
-                    .map(region -> cacheSettings.getProperty(region))
-                    .filter(Objects::nonNull)
-                    .collect(Collectors.toSet());
-
-            int length = INFINISPAN_CONFIG_RESOURCE_PROP.length();
-            String customRegionPrefix = INFINISPAN_CONFIG_RESOURCE_PROP.substring(0, length - 3) + mutableProperties.getProperty(AvailableSettings.CACHE_REGION_PREFIX, "");
-            String customRegionSuffix = INFINISPAN_CONFIG_RESOURCE_PROP.substring(length - 4, length);
-
-            // Collect distinct cache configurations for custom regions
-            Set<String> customRegionConfigs = mutableProperties.stringPropertyNames().stream()
-                    .filter(name -> name.startsWith(customRegionPrefix) && name.endsWith(customRegionSuffix))
-                    .map(name -> mutableProperties.getProperty(name))
-                    .filter(config -> !standardRegionConfigs.contains(config))
-                    .collect(Collectors.toSet());
-
-            if (!customRegionConfigs.isEmpty()) {
-                cacheSettings.setProperty(CUSTOM, String.join(" ", customRegionConfigs));
-            }
+            cacheSettings.setProperty(CONTAINER, container);
+            cacheSettings.setProperty(CACHES, String.join(" ", findCaches(mutableProperties)));
 
             Notification.addCacheDependencies(Classification.INFINISPAN, cacheSettings);
         }
     }
+
+    public static Set<String> findCaches(Properties properties) {
+        Set<String> caches = new HashSet<>();
+
+        caches.add(properties.getProperty(ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(IMMUTABLE_ENTITY_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
+        caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE));
```

Comment:
Please remove one of the duplicate calls to caches.add(properties.getProperty(COLLECTION_CACHE_RESOURCE_PROP, DEF_ENTITY_RESOURCE))